tests/integration/kubernetes/gha-run.sh

```diff
@@ -268,14 +268,15 @@ function run_tests() {
 	# and enable cri plugin in containerd config.
 	# TODO: enable testing auto-generated policy for other types of hosts too.
 
-	if [ "${KATA_HOST_OS}" = "cbl-mariner" ]; then
+	if [ "${KATA_HOST_OS}" = "cbl-mariner" ] || [ "${KATA_HYPERVISOR}" = "qemu-tdx" ] ; then
 
 		export AUTO_GENERATE_POLICY="yes"
-
-		# set default containerd config
-		sudo containerd config default | sudo tee /etc/containerd/config.toml > /dev/null
-		echo "containerd config has been set to default"
-		sudo systemctl restart containerd && sudo systemctl is-active containerd
+		if [ "${GENPOLICY_PULL_METHOD}" = "containerd" ]; then
+			# set default containerd config
+			sudo containerd config default | sudo tee /etc/containerd/config.toml > /dev/null
```

Comment:
Same here about the containerd configuration.  When dealing with TDX, we're using k3s, and the configuration for k3s lives in `/var/lib/rancher/k3s/agent/etc/containerd/config.toml.tmpl`, and if you override that, we'd end up facing issues.

Reply:
Hey @fidencio. But TDX, at least for now, won't use containerd pull method. It will use the [default](https://github.com/kata-containers/kata-containers/blob/main/tests/integration/kubernetes/gha-run.sh#L37) (oci-distribution-client), so TDX doesn't need and won't hit this code path. Here is a passing [test run](https://github.com/kata-containers/kata-containers/actions/runs/9423256330/job/25961925186?pr=9606#step:9:693) on TDX from a bigger draft PR with these changes. We can worry later about containerd config adjustments for k3s if and when we want to add a TDX + containerd pull entry to the test matrix. Thoughts?